packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -245,6 +245,12 @@ DataGridPremiumRaw.propTypes = {
    * @default false
    */
   disableDensitySelector: PropTypes.bool,
+  /**
+   * If `true`, `eval()` is not used for performance optimization.
+   * @default false
```

Comment:
We have reviewed the eval'd code in #9635 and #9120 and we haven't found any security risk. In particular, we ensure that no user inputted string can end up in the eval'd code without being properly escaped (aka no XSS). There are only two user inputs, `applier.item.id` and `applier.item.field`. We pass them both through `JSON.stringify`, which by the spec returns a valid & properly escaped string representation for JS/JSON.

If we do find a security risk, then it would be best to simply not include the feature.

Reply:
Alright 👍